sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -630,6 +630,15 @@ object RapidsConf {
     .booleanConf
     .createWithDefault(false)
 
+  val ENABLE_EXTENDED_YEAR_PARSING = conf("spark.rapids.sql.parseExtendedYears.enabled")
+      .doc("SPARK-35780 went into Spark 3.2.0+ and extended parsing of years in dates and " +
```

Comment:
Nit: Is the Spark JIRA too much detail for users here?  Wondering if this should be simplified to something like:
```
Spark 3.2+ parses the full range of possible year values in dates and timestamps beyond the typical 4-digit year. When this is enabled it is assumed all year values are not using the extended syntax and thus can be processed on the GPU which does not yet support extended range year values.
```

Also wondering if the configuration name is a bit misleading.  It's not so much enabling parsing of extended years as it is the user confirming there are no extended year values in the data.  In that sense it seems similar to the hasNans config.

Reply:
Do you have abetter name?

Reply:
`spark.rapids.sql.noExtendedYearValues` sounds like a better name to me, since it's focusing on the user validating there are no extended year values that need to be dealt with rather than implying there's extended year parsing that is disabled by default.